ironfish/src/network/peers/addressManager.ts

```diff
@@ -12,71 +13,113 @@ import { PeerManager } from './peerManager'
  * and provides functionality for persistence of said data.
  */
 export class AddressManager {
+  LIMIT = 50
+
   hostsStore: HostsStore
   peerManager: PeerManager
+  peerIdentityMap: Map<Identity, PeerAddress>
 
   constructor(hostsStore: HostsStore, peerManager: PeerManager) {
     this.hostsStore = hostsStore
     this.peerManager = peerManager
-  }
+    // load prior peers from disk
+    this.peerIdentityMap = new Map<string, PeerAddress>()
+    const currentTime = Date.now()
+    let priorPeers = this.hostsStore.getArray('priorPeers')
 
-  get priorConnectedPeerAddresses(): ReadonlyArray<Readonly<PeerAddress>> {
-    return this.hostsStore.getArray('priorPeers')
-  }
-
-  /**
-   * Returns a peer address for a disconnected peer by using current peers to
-   * filter out peer addresses. It attempts to find a previously-connected
-   * peer address that is not part of an active connection.
-   */
-  getRandomDisconnectedPeerAddress(peerIdentities: string[]): PeerAddress | null {
-    if (this.priorConnectedPeerAddresses.length === 0) {
-      return null
+    // If there are more than 50 peers, we remove
+    // extra peers from the list. This should only happen during
+    // the first time the node is started after this change is implemented.
+    if (priorPeers.length > this.LIMIT) {
+      priorPeers = priorPeers.slice(0, this.LIMIT)
     }
 
-    const currentPeerIdentities = new Set(peerIdentities)
+    for (const peer of priorPeers) {
+      if (peer.identity === null || peer.address === null || peer.port === null) {
+        continue
+      }
+
+      // Backwards compatible change: if lastAddedTimestamp is undefined or null,
+      // set it to the current time.
+      if (peer.lastAddedTimestamp === undefined) {
+        peer.lastAddedTimestamp = currentTime
```

Comment:
In `addPeer` we're using this `lastAddedTimestamp` to indicate when the last time these peers were connected to via websocket. These peers though we have no record of them connecting via websocket. Do we want to set this value here?

Reply:
Alternatively I think we could just rely on the calls to `peerManager.connectToWebSocketAddress` on startup to add these legacy peers into the address manager if we are able to connect to them via websocket

Reply:
I think we can set the timestamp to 0. 

Then when the first loop executes, the peerManager.connectToWebSocketAddress will trigger the events to update the timestamp. I just tested this locally to confirm

Reply:
My take is that they will get filtered out eventually as we create more connections with peers with using websocket outbound.